File: Dot examples/ASP Book Examples/Book Examples/Chapter24/CS/App_Code/SiteProfile.cs
using System;
using System.Web.Profile;

public class SiteProfile : ProfileBase
{
    private string _firstName = "Your First Name";
    private string _lastName = "Your Last Name";

    [SettingsAllowAnonymous(true)]
    public string FirstName
    {
        get { return _firstName; }
        set { _firstName = value; }
    }

    [SettingsAllowAnonymous(true)]
    public string LastName
    {
        get { return _lastName; }
        set { _lastName = value; }
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter24/VB/App_Code/SiteProfile.vb
Imports System
Imports System.Web.Profile

Public Class SiteProfile
    Inherits ProfileBase

    Private _firstName As String = "Your First Name"
    Private _lastName As String = "Your Last Name"

    <SettingsAllowAnonymous(True)> _
    Public Property FirstName() As String
        Get
            Return _firstName
        End Get
        Set(ByVal Value As String)
            _firstName = value
        End Set
    End Property

    <SettingsAllowAnonymous(True)> _
    Public Property LastName() As String
        Get
            Return _lastName
        End Get
        Set(ByVal Value As String)
            _lastName = value
        End Set
    End Property
End Class

